rj_gameplay/rj_gameplay/skill/mark.py

```diff
@@ -11,9 +11,36 @@
 import stp.skill as skill
 import stp.role as role
 import stp.action as action
+from rj_gameplay.action import move
 from stp.skill.action_behavior import ActionBehavior
-from stp.skill.cost_behavior import CostBehavior
 import stp.rc as rc
+import stp.utils.constants as constants
+
+from rj_geometry_msgs.msg import Point, Segment
+
+def get_mark_point(target_robot_id: int, world_state: rc.WorldState):
+    # workaround for non-working CostBehavior: 
+    # initialize move action, update target point every tick (target point being opponent robot pos)
+
+    # TODO: use mark_heuristic & CostBehavior to handle marking
+    # argument: mark_heuristic: Callable[[np.array], float]
+    # > self.mark_heuristic = mark_heuristic
+
+    # dist away from target_robot to mark
+    # TODO: add to global param server
+    SAG_DIST = constants.Robot.RADIUS * 0.5
+
+    # find point between ball and target robot that leaves SAG_DIST between edges of robots
+    ball_pos = world_state.ball.pos
+    opp_pos = world_state.their_robots[target_robot_id].pose[0:2]
+
+    mark_dir = (ball_pos - opp_pos) / np.linalg.norm(ball_pos - opp_pos)
+    mark_dist = mark_dir * (2.0 * constants.Robot.RADIUS + SAG_DIST)
+
+    # if ball too close to robot, can't mark successfully
+    if np.linalg.norm(mark_dist) < 2.0 * constants.Robot.RADIUS:
+        return None
```

Comment:
I don't think it should be None, you should probably return a point in front of the robot, like trying to block a pass or shot coming from them.